test/extended/images/imagestreamimport.go

```diff
@@ -280,49 +318,51 @@ func deployEphemeralImageRegistry(oc *exutil.CLI, proj *projectv1.Project) error
 		metav1.CreateOptions{},
 	)
 	if err != nil {
-		return err
+		return secret, nil, nil, fmt.Errorf("error creating registry deploy: %s", err)
 	}
 
-	// awaits for deployment to rollout.
-	if err := wait.Poll(time.Second, 5*time.Minute, func() (stop bool, err error) {
-		deploy, err := oc.AdminKubeClient().AppsV1().Deployments(proj.Name).Get(
+	t.Log("awaiting for registry deployment to rollout")
+	if err := wait.Poll(5*time.Second, 5*time.Minute, func() (stop bool, err error) {
+		deploy, err := oc.AdminKubeClient().AppsV1().Deployments(oc.Namespace()).Get(
 			context.Background(), deploy.Name, metav1.GetOptions{},
 		)
 		if err != nil {
 			return false, err
 		}
-		return deploy.Status.AvailableReplicas == replicas, nil
+		succeed := deploy.Status.AvailableReplicas == replicas
+		if !succeed {
+			t.Logf("registry deployment not ready yet, status: %+v", deploy.Status)
+		}
+		return succeed, nil
 	}); err != nil {
-		return err
+		e2e.DumpAllNamespaceInfo(oc.AdminKubeClient(), oc.Namespace())
```

Comment:
You don't need to call this function for `oc.Namespace()`. This repository uses [Ginkgo](https://onsi.github.io/ginkgo/#structuring-your-specs) that makes tests more declarative. The namespace is created in `BeforeEach` and is deleted in `AfterEach`, all necessary debug information for the namespace is generated by `AfterEach`. I guess these tests also should use `BeforeEach` and `AfterEach` and shouldn't use `g.GinkgoTInterface` and the latter is a helper for legacy code.

Reply:
PTAL